core/kernel/ldelf_syscalls.c

```diff
@@ -144,8 +144,21 @@ TEE_Result ldelf_sc_open_bin(const TEE_UUID *uuid, size_t uuid_size,
 		DMSG("res=0x%x", res);
 		if (res != TEE_ERROR_ITEM_NOT_FOUND &&
 		    res != TEE_ERROR_STORAGE_NOT_AVAILABLE)
-			break;
+			goto ts_found;
 	}
+
+	SCATTERED_ARRAY_FOREACH(binh->op, sp_stores, struct ts_store_ops) {
```

Comment:
This caller of this function knows if he's expecting a TA or a SP. So instead of relying on the caller being able to tell if it got an unexpected binary I think we should test that here directly instead. This way we can also separate the UUID name spaces of TAs and SPs so there can't be any collisions.

Reply:
Yes, makes sense. I decided to remove this commit from the PR, because this is not really part of the syscall separation, but a next step enhancement. I'll look into this and address it in a new PR.